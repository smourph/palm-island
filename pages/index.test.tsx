import { render, screen } from '@testing-library/react';
import Index from './index';

describe('Index', () => {
  it('should render', () => {
    const { container } = render(<Index/>);

    expect(container).toMatchSnapshot();
  });

  it('should display board', () => {
    render(<Index/>);

    const board = screen.getByRole('table');
    expect(board).toBeInTheDocument();
    expect(board.id).toBe('board');

    const boardContent = screen.getByRole('rowgroup');
    expect(boardContent).toBeInTheDocument();

    const boardCells = screen.getAllByRole('cell');
    expect(boardCells).toHaveLength(9);
  });

  it('should display footer', () => {
    render(<Index/>);

    const footer = screen.getByRole('contentinfo');
    expect(footer).toBeInTheDocument();
    expect(footer.textContent).toBe('Palm Island game');
  });
});
