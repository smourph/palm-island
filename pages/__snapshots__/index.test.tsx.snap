// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render 1`] = `
<div>
  <div>
    <div
      class="bgio-client"
    >
      <div>
        <table
          id="board"
        >
          <tbody>
            <tr>
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
            </tr>
            <tr>
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
            </tr>
            <tr>
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
              <td
                style="border: 1px solid #555; width: 50px; height: 50px; line-height: 50px; text-align: center;"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <footer>
      <hr />
      <span>
        Palm Island game
      </span>
    </footer>
  </div>
</div>
`;
